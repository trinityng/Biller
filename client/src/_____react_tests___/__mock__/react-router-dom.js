import React from 'react';
// Just render plain div with its children
const rrd = require('react-router-dom');

rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;