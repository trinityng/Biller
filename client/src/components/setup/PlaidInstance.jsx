import React from 'react';
import { Statistic, Row, Col, Card } from 'antd';

const PlaidInstance = props => {
  return (
    <Card>
      <Card>
        <Row gutter={19}>
          <Col span={12}>
            <Statistic title="Account" value={props.bankName} />
          </Col>
          <Col span={12}>
            <Statistic title="Account Type" value={props.accountNum} />
          </Col>
        </Row>
      </Card>
    </Card>
  );
};

export default PlaidInstance;
