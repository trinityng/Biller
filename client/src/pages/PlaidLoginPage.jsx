import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Redirect, withRouter } from 'react-router-dom';

import { plaidLogin } from '../api/plaid.api';
import SetupAccount from '../components/setup/SetupAccounts';
import { updateProfile } from '../store/actions/auth.action';
import { USER_TOKEN, ACCOUNTS_INFO } from '../constants';

import './Pages.css';

class PlaidLoginPage extends Component {
  constructor(props) {
    super(props);

    this.onSuccess = this.onSuccess.bind(this);
  }

  componentDidMount() {
    document.body.style.backgroundColor = '#F0F2F5';
  }

  onExit() {}

  onSuccess(public_token, metadata) {
    console.log(public_token);
    const { user, plaid } = this.props;
    const { email, token } = user;
    console.log('[SetupForm] user: ', user);

    this.props.plaidLogin({
      email,
      token,
      public_token,
    });

    const previousForms = JSON.parse(localStorage.getItem(ACCOUNTS_INFO)) || [];
    const forms = [
      ...previousForms,
      {
        accountNum: metadata.accounts.length,
        bankName: metadata.institution.name,
      },
    ];

    console.log('plaid.errors: ' + plaid.errors);
    if (!plaid.errors) {
      this.props.updateProfile({ has_profile: true });
      localStorage.setItem(ACCOUNTS_INFO, JSON.stringify(forms));
    }
  }

  render() {
    const token = localStorage.getItem(USER_TOKEN);
    const { user } = this.props;
    const forms = JSON.parse(localStorage.getItem(ACCOUNTS_INFO)) || [];

    console.log('forms: ', forms);
    return token ? (
      <SetupAccount
        forms={forms}
        has_profile={user.has_profile}
        onSuccess={this.onSuccess}
      />
    ) : (
      <Redirect to="/login" />
    );
  }
}

function mapStateToProps(state) {
  const { user, plaid } = state;
  return {
    user,
    plaid,
  };
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators({ updateProfile, plaidLogin }, dispatch);
}

export default connect(
  mapStateToProps,
  matchDispatchToProps
)(withRouter(PlaidLoginPage));
