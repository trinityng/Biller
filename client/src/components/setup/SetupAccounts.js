import React from 'react';

import { Link } from 'react-router-dom';
import { PlaidLink } from 'react-plaid-link';
import { Layout, List, Button } from 'antd';
import { PlusOutlined, ArrowRightOutlined } from '@ant-design/icons';
import {
  PLAID_PRODUCT,
  PLAID_SB_ENV,
  //   PLAID_DEV_ENV,
  PLAID_PUlLIC_KEY,
} from '../../constants';

import PlaidInstance from './PlaidInstance';

const { Content } = Layout;

const SetupAccounts = props => {
  return (
    <div style={{ margin: '5vh' }}>
      <Layout className="setup-form">
        <Content style={{ marginBottom: '54vh' }}>
          <List
            grid={{
              gutter: [48, 16],
              xs: 1,
              sm: 1,
              md: 1,
              lg: 2,
              xl: 2,
              xxl: 2,
            }}
            dataSource={props.forms}
            renderItem={item => (
              <List.Item>
                <PlaidInstance
                  bankName={item.bankName}
                  accountNum={item.accountNum}
                  onDelete={this.onDelete}
                />
              </List.Item>
            )}
          />
          <div className="plaid-div">
            <PlaidLink
              style={{
                padding: '14px',
                fontSize: '16px',
                cursor: 'pointer',
                border: 'dashed 2px rgb(120, 120, 120)',
              }}
              clientName="SJSU-Biller"
              env={PLAID_SB_ENV}
              product={PLAID_PRODUCT}
              publicKey={PLAID_PUlLIC_KEY}
              onSuccess={props.onSuccess}
              //   onExit={this.onExit}
            >
              <PlusOutlined /> Connect with Plaid
            </PlaidLink>
            {props.has_profile && (
              <Link to="/dashboard">
                <Button
                  type="primary"
                  icon={<ArrowRightOutlined />}
                  size="large"
                  className="connect-dashboard-btn"
                >
                  Proceed to Dashboard
                </Button>
              </Link>
            )}
          </div>
        </Content>
      </Layout>
    </div>
  );
};

export default SetupAccounts;
