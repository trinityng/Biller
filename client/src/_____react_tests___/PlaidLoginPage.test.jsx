import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import {mount, shallow, configure } from 'enzyme';
import { MemoryRouter, Router } from "react-router-dom";
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

import PlaidLoginPage from '../pages/RegisterPage';
import SetupAccounts from '../components/setup/SetupAccounts';
import PlaidInstance from '../components/setup/PlaidInstance';


Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

configure({adapter: new Adapter()});

describe('plaid login page / components', () => {
  let initialState = {};
  let store;

  beforeEach(() => {
    store = mockStore(initialState)
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/setup']}>
          <PlaidLoginPage />
        </MemoryRouter>,
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe("when user is not logged in", () => {
    it('should redirect the user to login page', () => {
     
    });
  });

  describe("when user is logged in", () => {
    it('should allow user to connect to plaid API', () => {
     
    });

    it('should show a component to the user after connection with plaid successful', () => {
     
    });
  });

  describe("when user has at least one connected account", () => {
    it('should allow user to go to dashboard', () => {
     
    });
  });
});