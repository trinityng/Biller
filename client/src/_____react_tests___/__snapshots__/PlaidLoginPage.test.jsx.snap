// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plaid login page / components should render correctly 1`] = `ReactWrapper {}`;
